src/main/java/school/hei/tsinjo/model/psp/vola/VolaPsp.java

```diff
@@ -31,26 +32,59 @@ public Payment get(String tsinjoId, PspType pspType, String pspId, String email)
 
   private Payment toPayment(
       String tsinjoId, school.hei.tsinjo.model.psp.vola.api.gen.client.model.Payment volaPayment) {
+    if (tsinjoId == null) {
+      throw new IllegalArgumentException("tsinjoId cannot be null");
+    }
+    if (volaPayment == null) {
+      throw new VolaPaymentNullException("Vola payment is null for tsinjoId: " + tsinjoId);
+    }
 
-    var volaPspPayment = volaPayment.getPspPayment();
+    PspPayment volaPspPayment = volaPayment.getPspPayment();
 
-    // ✅ gestion du null pour éviter le NPE
     Instant lastVerificationInstant =
         Optional.ofNullable(volaPayment.getLastPspVerificationInstant())
             .map(Date::toInstant)
             .orElse(null);
 
+    Instant chosenCreation =
+        determineCreationInstant(volaPayment, volaPspPayment, lastVerificationInstant);
+
     return new Payment(
         tsinjoId,
-        volaPspPayment.getAmount(),
-        toPspType(volaPspPayment.getPspType()),
-        volaPspPayment.getId(),
+        (volaPspPayment == null ? null : volaPspPayment.getAmount()),
```

Comment:
Still bad. More readable would be return volaPspPayment == null ? new Payment(...) : new Payment(...))